docs/readmes/architecture/agw_overview.md

```diff
@@ -0,0 +1,59 @@
+
+10. **Policydb** - PolicyDB is the service that supports static PCRF rules. This service runs in both the AGW and the orchestrator. Rules managed through the rest API are streamed to the policydb instances on the AGW. Sessiond ensures these policies are implemented as specified.
+
+11. **Dpid** - Deep packet inspection service to enforce policy rules.
+
+12. **Pipelined** - Pipelined is the control application that programs the OVS openflow rules. Pipelined is a set of services that are chained together. These services can be chained and enabled/disabled through the REST API. If pipelined is restarted, users service will be affected.
+
+13. **Eventd** - Service that acts like an intermediary for different magma services, using the service303 interface, it will receive and push the generated registered events to the td-agent-bit service on the gateway, so these can be then later sent to Orchestrator. These events will be sent to ElasticSearch where they can be queried. 
+
+14. **Monitord** - a dynamic service that monitors the CPEs connected to the AGW. It will send ICMP pings to the CPEs connected to the gateway and report if they are active. 
+
+15. **SMSd** - service that functions as the AGW interface that will sync the SMS information with Orc8r. 
+
+16. **Td-agent-bit** - consists of Fluentbit and is run as a dynamic service on the AGW. To use td-agent-bit the user must modify the gateway magmad configuration. It is used for log aggregation and event logging where it takes input from syslog and the events service and forwards the output to the Orc8r. It is received on the Orc8r by Fluentd where it is stored in Elasticsearch. 
+
+17. **Ctraced** - Service used for managing call tracing on the AGW. The Tshark tool is used for packet capture and filtering. Packet captures are sent back to the orc8r and viewable on the NMS. Preferred usage for call tracing is through the NMS.
+
+18.  **Health** - Health checker service that verifies the state on MME, mobilityd, sessiond and pipelined and cleans corrupt state if necessary.
+
+19. **Control Proxy** - Control proxy manages the network transport between the gateways and the controller. It additionally provides the following functionality:
+	* Control proxy abstract the service addressability, by providing a service registry which maps a user addressable name to its remote IP and port.
+	* All traffic over HTTP/2, and are encrypted using TLS. The traffic is routed to individual services by encoding the service name in the HTTP/2 :authority: header.
+	* Individual GRPC calls between a gateway and the controller are multiplexed over the same HTTP/2 connection, and this helps to avoid the connection setup time per RPC call.
```

Comment:
Could you check with @pshelar on envoy service?

Reply:
@pshelar / @ulaskozat what are your thoughts on adding this to cover the header enrichment services:

"Header Enrichment Services - there are two services within header enrichment: the envoy controller and envoy_dp. Envoy functions as the de facto HTTP proxy used for service mesh and can be programmed via the gRPC API. Additionally, the Envoy dataplane can scale up and scale down according to resources on AGW. The envoy controller is programmed by PipelineD according to the set of rules in place. "